File: backend/app/request/admin/asset.go
package admin

import (
	"time"

	"github.com/phongsakk/finn4u-back/utils"
)

type DoAppraisal struct {
	PriceAppraisal float64   `json:"price_appraisal" validate:"required"`
	CollateraPrice float64   `json:"collateral_price" validate:"required"`
	Duration       int       `json:"duration" validate:"required"`
	DisplayImages  *[]uint   `json:"display_images" validate:"omitempty"`
	NewImages      *[]string `json:"new_images" validate:"omitempty"`
	Tags           *[]uint   `json:"tags" validate:"omitempty"`
	IsPublished    *bool     `json:"is_published" validate:"omitempty"`
	Status         *uint     `json:"status" validate:"omitempty"`
	Auction        *Auction  `json:"auction" validate:"omitempty"`
}

type Auction struct {
	FromDate time.Time `json:"from_date" validate:"required"`
	ToDate   time.Time `json:"to_date" validate:"required"`
	FromTime string    `json:"from_time" validate:"required"`
	ToTime   string    `json:"to_time" validate:"required"`
	MaxTax   float32   `json:"max_tax" validate:"required"`
}

func (r *DoAppraisal) Validated() error {
	return utils.Validate(r)
}

File: admin/src/models/asset.ts
export type ConsignParam = {
  id: number;
  status: boolean;
};

export type Estimateprice = {
  id: number;
  status: number;
  is_published: boolean;
  images: assetImages[];
  tags: tags[];
};

export type tags = {
  id: string;
  name: string;
};
export type assetImages = {
  id: number;
  name: string;
};

export type DoAppraisal = {
  price_appraisal: number;
  collateral_price: number;
  duration: number;
  display_images?: number[];
  tags?: number[];
  is_published?: boolean;
  status: number;
  auction?: Auction;
};

type Auction = {
  from_date: Date;
  to_date: Date;
  from_time: string;
  to_time: string;
  max_tax: number;
};
